.github/workflows/ci.yml

```diff
@@ -86,38 +93,50 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get install gcc-multilib
-      - name: Test almost no features
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          target: ${{ matrix.target }}
+          override: true
+      - name: Cache Builds
+        uses: Swatinem/rust-cache@v1
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Test (Minimal)
         uses: actions-rs/cargo@v1
-        if: matrix.features == 'none'
+        if: matrix.features == 'minimal'
         with:
           command: test
-          args: --target ${{ matrix.target }} --no-default-features --features "std cargo" -p clap:3.0.0-beta.5
-      - name: Test all features
+          args: --target ${{ matrix.target }} ${{ needs.args.outputs.minimal }}
+      - name: Test (Full)
         uses: actions-rs/cargo@v1
-        if: matrix.features == 'all'
+        if: matrix.features == 'full'
         with:
           command: test
-          args: --target ${{ matrix.target }} --features "wrap_help yaml regex unstable-replace unstable-multicall unstable-grouped"
+          args: --target ${{ matrix.target }} ${{ needs.args.outputs.full }}
```

Comment:
In #2901, you mentioned cherry-picking changes over.  With the simplified pipeline that offers, I had switched us over to separating compilation from testing, like `ci-pr.yml` (which I originally didn't do to `ci.yml` because the logic was complicated as-is).

Where do you see the trade off of that now since this doesn't include cherry-picking that change and doesn't comment on it?